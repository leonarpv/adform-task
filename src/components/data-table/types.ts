import { ReactNode } from 'react'

export type ColumnDefinition<TSource extends {}> = {
    accessor: (source: TSource) => ReactNode
    title: string
}
