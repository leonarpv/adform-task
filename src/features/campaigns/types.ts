export type Campaign = {
    id: number
    name: string
    startDate: string
    endDate: string
    Budget: number
    userId: number
}

export type CampaignUser = {
    id: number
    name: string
}

export type CampaignWithUser = {
    user: CampaignUser | null
} & Campaign

export type CampaignFilters = {
    name?: string
    startDate?: string
    endDate?: string
}
