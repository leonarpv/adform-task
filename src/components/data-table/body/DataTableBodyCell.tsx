import React from 'react'
import { ColumnDefinition } from '../types'

type Props<TSource> = {
    column: ColumnDefinition<TSource>
    sourceEntry: TSource
}

const DataTableBodyCell = <TSource extends {}>({
    column,
    sourceEntry,
}: Props<TSource>) => {
    const element = column.accessor(sourceEntry)

    return (
        <td className="data-table__cell data-table__cell--body">{element}</td>
    )
}

export default DataTableBodyCell
