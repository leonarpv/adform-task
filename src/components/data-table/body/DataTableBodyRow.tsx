import React from 'react'
import { ColumnDefinition } from '../types'
import DataTableBodyCell from './DataTableBodyCell'

type Props<TSource> = {
    columns: ColumnDefinition<TSource>[]
    sourceEntry: TSource
}

const DataTableBodyRow = <TSource extends {}>({
    columns,
    sourceEntry,
}: Props<TSource>) => {
    const cells = columns.map((c, index) => (
        <DataTableBodyCell key={index} column={c} sourceEntry={sourceEntry} />
    ))

    return <tr className="data-table__row">{cells}</tr>
}

export default DataTableBodyRow
