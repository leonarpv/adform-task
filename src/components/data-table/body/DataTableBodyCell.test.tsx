import { shallow } from 'enzyme'
import React from 'react'
import { ColumnDefinition } from '../types'
import DataTableBodyRow from './DataTableBodyRow'
import { Mock } from 'ts-mockery'
import DataTableBodyCell from './DataTableBodyCell'

describe('DataTableBodyCell', () => {
    it('should render properly', () => {
        const text = 'text'
        const column = Mock.of<ColumnDefinition<string>>({
            accessor: () => text,
        })
        const sourceEntry = 'sourceEntry'

        const wrapper = shallow(
            <DataTableBodyCell column={column} sourceEntry={sourceEntry} />
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('should render table head', () => {
        const text = 'text'
        const column = Mock.of<ColumnDefinition<string>>({
            accessor: () => text,
        })
        const sourceEntry = 'sourceEntry'

        const wrapper = shallow(
            <DataTableBodyCell column={column} sourceEntry={sourceEntry} />
        )

        expect(
            wrapper.find('.data-table__cell.data-table__cell--body').exists()
        ).toBe(true)
        expect(
            wrapper.find('.data-table__cell.data-table__cell--body').text()
        ).toBe(text)
    })
})
