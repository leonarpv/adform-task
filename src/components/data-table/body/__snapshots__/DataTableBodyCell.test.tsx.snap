// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableBodyCell should render properly 1`] = `ShallowWrapper {}`;
